app/services/hyrax/solr_query_service.rb

```diff
@@ -3,7 +3,7 @@ module Hyrax
   ##
   # Methods in this class are providing functionality previously supported by ActiveFedora::SolrQueryBuilder.
   # It includes methods to build and execute a query.
-  class SolrQueryService
+  class SolrQueryService < ::SearchBuilder
```

Comment:
Inheriting from `SearchBuilder` is required to have access to the private method `gated_discovery_filters`.